LoopKit/CarbKit/CarbStore.swift

```diff
@@ -229,7 +229,28 @@ public final class CarbStore: HealthKitSampleStore {
             }
 
             UserDefaults.standard.purgeLegacyCarbEntryKeys()
-            
+
+            // Migrate all DeletedCarbObject to updated CachedCarbObject
+            self.cacheStore.managedObjectContext.performAndWait {
```

Comment:
Since part of this migration is to delete the DeletedCarbObject from the database (line 244), any future runs will find no DeletedCarbObjects in the database and will be a no-op effectively. I left is how it is here for two reasons:
- We'd need to keep track some place that the migration has been performed, but that's just storing it someplace else. Pretty much the same overhead as doing a fetch for any DeletedCarbObjects and getting no objects back.
- I considered the use case of a DIYer upgrading to a newer version, testing it out for a bit, downgrading, and then later upgrading again. The code as-is properly handles this case (rare, but I'm sure someone will do it).